src/calculus/slicedSeparableSum.jl

```diff
@@ -32,7 +32,7 @@ function SlicedSeparableSum{S <: AbstractArray, T <: AbstractArray}(fs::S, idxs:
     error("size(fs) must coincide with size(idxs)")
   else
     for k in eachindex(idxs)
-      cond = [typeof(t) <: Integer || typeof(t) <: AbstractArray{Integer,1} || typeof(t) <: Colon || typeof(t) <: Range for t in idxs[k]]
+      cond = [typeof(t) <: Integer || typeof(t) <: AbstractArray{Int,1} || typeof(t) <: Colon || typeof(t) <: Range for t in idxs[k]]
```

Comment:
The problem was that in Julia `S <: T` does not imply `AbstractArray{S} <: AbstractArray{T}`.

I guess this whole check should somehow go in the type parameters constraints (line 30). However, to keep it like this for now, it's probably better to have something like
```
typeof(t) <: AbstractArray && eltype(t) <: Integer
```

Reply:
The reason being that `Int <: Integer`

Reply:
if we want to do these checks directly on the constructor (and I am in favour of this) I think it would be better to use `Tuples` instead of `Arrays` both for `idxs` and `fs`! 

That is because 
```julia 
julia> typeof([1,:])
Array{Any,1}
```
while 
```
julia> typeof((1,:))
Tuple{Int64,Colon} 
```
which has the advantage being capable of doing this:
```
julia> typeof((1,:)) <: NTuple{2,Union{Integer,Colon}}
true
```

Reply:
You know better how `SlicedSeparableSum` works, you have carta bianca ;-)